src/App.tsx

```diff
@@ -7,36 +7,85 @@ import classNames from 'classnames';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { getPostsByUserId } from './api/posts';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [hasError, setHasError] = useState(false);
+
+  useEffect(() => {
+    if (selectedUser) {
+      setIsLoading(true);
+      setHasError(false);
+      setSelectedPost(null);
+
+      getPostsByUserId(selectedUser.id)
+        .then(res => {
+          if ('error' in res) {
+            throw new Error();
+          }
```

Comment:
no need to throw error here. If there will be an error in response catch branch will work